crates/re_renderer/src/view_builder.rs

```diff
@@ -314,89 +381,36 @@ impl ViewBuilder {
-                    //                                              = tan(FOV / 2) * 2.0 / resolution
-                    let pixel_world_size_from_camera_distance =
-                        tan_half_fov.y * 2.0 / config.resolution_in_pixel[1] as f32;
-
-                    (
-                        projection_from_view,
-                        tan_half_fov,
-                        pixel_world_size_from_camera_distance,
-                    )
-                }
-                Projection::Orthographic {
-                    camera_mode,
-                    vertical_world_size,
-                    far_plane_distance,
-                } => {
-                    let aspect_ratio =
-                        config.resolution_in_pixel[0] as f32 / config.resolution_in_pixel[1] as f32;
-                    let horizontal_world_size = vertical_world_size * aspect_ratio;
-                    // Note that we inverse z (by swapping near and far plane) to be consistent with our perspective projection.
-                    let projection_from_view = match camera_mode {
-                        OrthographicCameraMode::NearPlaneCenter => glam::Mat4::orthographic_rh(
-                            -0.5 * horizontal_world_size,
-                            0.5 * horizontal_world_size,
-                            -0.5 * vertical_world_size,
-                            0.5 * vertical_world_size,
-                            far_plane_distance,
-                            0.0,
-                        ),
-                        OrthographicCameraMode::TopLeftCornerAndExtendZ => {
-                            glam::Mat4::orthographic_rh(
-                                0.0,
-                                horizontal_world_size,
-                                vertical_world_size,
-                                0.0,
-                                far_plane_distance,
-                                -far_plane_distance,
-                            )
-                        }
-                    };
-
-                    let tan_half_fov = glam::vec2(f32::MAX, f32::MAX);
-                    let pixel_world_size_from_camera_distance = vertical_world_size
-                        / config.resolution_in_pixel[0].max(config.resolution_in_pixel[1]) as f32;
-
-                    (
-                        projection_from_view,
-                        tan_half_fov,
-                        pixel_world_size_from_camera_distance,
-                    )
-                }
-            };
+        let aspect_ratio =
+            config.resolution_in_pixel[0] as f32 / config.resolution_in_pixel[1] as f32;
+
+        let projection_from_view = config
+            .projection_from_view
+            .projection_from_view(aspect_ratio);
+
+        let tan_half_fov = config.projection_from_view.tan_half_fov();
+
+        let pixel_world_size_from_camera_distance = match config.projection_from_view {
```

Comment:
Now that there's `projection_from_view` it would only be consistent to have a method on `Projection` that gives you `pixel_world_size_from_camera_distance` (you started it :P)

Reply:
counter argument is that this one is more complicated as it takes pan/zoom into account which is outside of `Projection`. Your call (:

Reply:
my call is already commited